certs/certs.go

```diff
@@ -29,4 +35,13 @@ type Repository interface {
 
 	// RetrieveBySerial retrieves a certificate for a given serial ID
 	RetrieveBySerial(ctx context.Context, ownerID, serialID string) (Cert, error)
+
+	//List the certifcate need to be renew before the expiry
+	ListExpiredCerts(ctx context.Context, timeBefore time.Duration, limit, offset uint64) (Page, error)
+
+	// AutoRetrieveByThingID  retrieves list of certificate for a given thing ID without owner ID , used  for AutoRenew process
+	AutoRetrieveByThingID(ctx context.Context, thingID string) (Cert, error)
```

Comment:
No need for the `Auto` prefix, it sounds misleading (I know we don't take owner ID into account, but that's fine because we do not have another repo method that does).

Reply:
The repo method ```RetrieveByThing``` do the same but with owner ID 
```go
RetrieveByThing(ctx context.Context, ownerID, thingID string, offset, limit uint64) (Page, error)
```
So to identify the function with name I kept like that way. I can't able to think for another name at that time , Could you suggest a better name or prefix ?

Reply:
You can make it a single method (`RetrieveByThingID`) and make it use userID _only if present_ (i.e. ignore if empty, the same way search and filtering by some criteria usually work). On the service level, you decide how to handle authorization.

Reply:
the  ```AutoRetrieveByThingID``` functions are no more need , the function are removed
the ```AutoRemoveByThingID``` function name changed to ```ThingCertsRevokeHandler```